### DISCUSSION TITLE
Promotions form further improvements

### DISCUSSION THREAD

**Discussion by tvdeyen** [solidusio/solidus#4570] [New Features or Ideas]
As a follow up to #1509 we should further improve the promotions form:

* [ ] Add hint to "advertise"
* [ ] Add hint to actions and rules
* [ ] Allow promotion action fields to update the promotion by enter key
* [ ] Align the layout
* [ ] Fix responsive layout
* [ ] Make the promotion code more prominent
* [ ] Link to promotion batch list from "This promotion uses x promotion codes" text
* [ ] Disallow the creation of promotion code batches for promotions without or only a single code

![new promotion align layout](https://cloud.githubusercontent.com/assets/42868/22584004/5a70aeaa-e9f0-11e6-94c8-35034e4dcdc4.png)
![solidus promotions form mobile](https://cloud.githubusercontent.com/assets/42868/22584003/5a6fe880-e9f0-11e6-81dd-cbacf077ac00.png)
![promotion code batches for promotion without code](https://cloud.githubusercontent.com/assets/42868/22584002/5a51d0d4-e9f0-11e6-9e66-9d433289b5cb.png)

**Comment by Mandily** [comment]
With regards to the point "Add hint to "advertise"" Instead of adding a hint, let's remove this from the admin. It's not worth the real estate or edge case of advertising a promotion on a specific page to keep. I had a chat with @jhawthorn and he's agreed. 

Two of these items are big enough I think they can be broken out into their own issues. I'll create a separate one that will include a wireframe to cover both of these: 
- Fix responsive layout
- Make the promotion code more prominent

I can come up with some copy for "Add hint to actions and rules" if you can refresh my memory on the specifics of this. 

I think the rest are pretty straight forward. If anyone has questions about what's left up there, please feel free to comment or ping me directly in Solidus Slack under the same handle.

**Comment by seand7565** [comment]
This issue is somewhat out of date. Current state of this issue, as far as I can tell:

`Add hint to "advertise"` - Advertise is no longer in master - at least, the checkbox is not there (The advertise boolean still exists though) ✅ 
`Add hint to actions and rules` - Hint exists on actions, not on rules. 🟨 
![image](https://user-images.githubusercontent.com/5720486/83359306-10e60a80-a33f-11ea-9129-1a54d64788e8.png)
`Allow promotion action fields to update the promotion by enter key` - This works ✅ 
`Align the layout` - The layout looks different now, not sure if it's what they meant? ❓ 
![image](https://user-images.githubusercontent.com/5720486/83359317-20655380-a33f-11ea-997f-3acf88c2961a.png)
`Fix responsive layout` - Again, not sure what was wrong with it in the first place, but here's the current layout on a small screen. ❓ 
![image](https://user-images.githubusercontent.com/5720486/83359338-3ecb4f00-a33f-11ea-9ebb-5336bb61cc09.png)
`Make the promotion code more prominent` - Not sure what was meant by this. ❓ 
`Link to promotion batch list from "This promotion uses x promotion codes" text` - Not done ❌ 
`Disallow the creation of promotion code batches for promotions without or only a single code` - Not entirely sure what is meant here, but you can still create promotion code batches for promotions that have a single promotion code, so I'm assuming it's not done! ❌ 

I'd love to hear some opinions on the unfinished items above - and whether we should extract them into a new, fresher issue - or just close this issue without replacement.
